gix-pack/src/index/access.rs

```diff
@@ -164,7 +164,7 @@ impl index::File {
                 let offset32_start = &self.data[self.offset_pack_offset_v2()..];
                 let pack_offset_64_start = self.offset_pack_offset64_v2();
                 offset32_start
-                    .chunks(N32_SIZE)
+                    .chunks_exact(N32_SIZE)
```

Comment:
Let's leave this at chunks as well, it should panic if the count is off (which it will).

Reply:
I've added an explicit length check instead. I think that makes intention clearer. It lowers risk of performing some work with side effects before panicking, and may be easier to diagnose bad inputs than panicking somewhere down the line.